### PR TITLE
Project setup

### PR DESCRIPTION
**Pull request by qroll**
### Build
Using rollup, with reference to the configuration from [Design System](https://github.com/LifeSG/react-design-system/blob/master/rollup.config.js)

- ESM/CJS
- Allow subpath imports for reduced bundle size on browsers
- For better tree-shaking results, to avoid using TS namespace (i.e. `export namespace`) in the submodules

```
import { JsonPath } from "@lifesg/ccube-lib"; // not recommended, no tree shaking
import { parse } from "@lifesg/ccube-lib/json-path"; // unused functions are not bundled
```

### Test
Using Jest

### Linting
Using eslint with new flat config. `eslintrc` will be officially deprecated in v9 (in alpha)

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import { nodeResolve } from "@rollup/plugin-node-resolve";
+import fs from "fs";
+import path from "path";
+import generatePackageJson from "rollup-plugin-generate-package-json";
+import peerDepsExternal from "rollup-plugin-peer-deps-external";
+import typescript from "rollup-plugin-typescript2";
+import pkg from "./package.json";
+
+export const plugins = [
+    peerDepsExternal(), // Exclude peer dependencies from bundle
```
Comment (weili-govtech) [inline rollup.config.js]: not needed as this library is not using peer dependencies, the package can be removed as well

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# dependencies
+node_modules
+
+# build
+dist
+*.tgz
+
+# misc
+junit.xml
```
Comment (weili-govtech) [inline .gitignore:9]: can add `.DS_Store`

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{
+    "compilerOptions": {
+        /** General settings **/
+        "target": "es6",
+        "module": "esnext",
+        "lib": ["ES2020", "ES2021"],
```
Comment (weili-govtech) [inline tsconfig.json]: what's the rationale of picking `ES2020` and `ES2021`?

File: eslint.config.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+module.exports = [
+    js.configs.recommended,
+    prettierRecommended,
+    {
+        ignores: ["node_modules", "dist"],
+    },
+    {
+        languageOptions: {
+            ecmaVersion: 2021,
+            globals: { ...globals.node, ...globals.browser, ...globals.es2015 },
```
Comment (weili-govtech) [inline eslint.config.js:16]: why do we use `es2015` for globals when the ecmaVersion is 2021?

**Review by weili-govtech — CHANGES_REQUESTED** [review]
it'll be good to add the github templates and the contributing doc too

File: eslint.config.js (hunk 4 above)
Comment (qroll) [reply eslint.config.js:16]: picking ecma 2021 as the baseline syntax

but using es6/es2015 as the minimum target for older browsers, so only es2015 globals will be available

Comment (qroll) [reply eslint.config.js:16]: hmm on that note, module in tsconfig.json should be updated to match ecmaVersion

File: rollup.config.js (hunk 1 above)
Comment (qroll) [reply rollup.config.js]: was thinking this will be for future use, but can be removed for now

File: tsconfig.json (hunk 3 above)
Comment (qroll) [reply tsconfig.json]: removed since no browser polyfills are in place

**Pull request merged**